test/unittests/state_transition.cpp

```diff
@@ -15,10 +15,29 @@ void state_transition::SetUp()
     expect.post[tx.sender].exists = true;
 }
 
+class TraceCapture
+{
+    std::streambuf* m_orig_rdbuf = nullptr;
+    std::ostringstream m_trace_stream;
+
+public:
+    TraceCapture() { m_orig_rdbuf = std::clog.rdbuf(m_trace_stream.rdbuf()); }
+    ~TraceCapture() { std::clog.rdbuf(m_orig_rdbuf); }
+    [[maybe_unused]] std::string get_capture() const { return m_trace_stream.str(); }
+};
+
 void state_transition::TearDown()
 {
     auto state = pre;
-    const auto res = evmone::state::transition(state, block, tx, rev, vm, block.gas_limit);
+    const auto trace = !expect.trace.empty();
+    auto& selected_vm = trace ? tracing_vm : vm;
+
+    /// Optionally enable trace capturing but keep the RAII object in scope.
+    std::optional<const TraceCapture> trace_capture;
+    if (trace)
+        trace_capture.emplace();
```

Comment:
could be something like
```suggestion
    const std::optional<const TraceCapture> trace_capture(trace ? TraceCapture{} : std::nullopt);
```

Reply:
`TraceCapture` is not movable because `std::ostringstream` is not (?).

Reply:
Ok, it seems it is movable. So maybe I need to enable move constructor in `TraceCapture`...

Reply:
I found a way, but it looks it relies on copy elision. It is nice but may backfire.